File: informe-3/melchor.cpp
#include "melchor.h"
/**
 * Declaracion de metodos de la clase Hombre
 * @param _nombre 
 */
void Hombre::setNombre(std::string nombre){
    this->nombre=nombre;
}
std::string Hombre::getNombre(){
    return this->nombre;
}
Hombre::Hombre(){}

/**
 Declaracion de metodos de la clase Mujer
 */
void Mujer::setNombre(std::string nombre){
    this->nombre=nombre;
}
std::string Mujer::getNombre(){
    return this->nombre;
}
void Mujer::setEdad(int edad){
    this->edad=edad;
}
int Mujer::getEdad(){
    return this->edad;
}
Mujer::Mujer(){}

/**
 * Metodos de la clase Estudiante
 * @param nombre
 */
Estudiante::Estudiante(){}
Estudiante::Estudiante(std::string nombre){
    this->nombre=nombre;
}
Estudiante::Estudiante(int nota){
    this->nota=nota;
}
Estudiante::Estudiante(std::string carnet, int nota){
    this->carnet=carnet;
    this->nota=nota;
}
void Estudiante::setNombre(std::string nombre){
    this->nombre=nombre;
}
std::string Estudiante::getNombre(){
    return this->nombre;
}
void Estudiante::setCarnet(std::string carnet){
    this->carnet=carnet;
}
std::string Estudiante::getCarnet(){
    return this->carnet;
}
void Estudiante::setNota(int nota){
    this->nota=nota;
}
int Estudiante::getNota(){
    return this->nota;
}


/**Declaracion de metodos de la clase Restaurante
 */
     //Constructor

void Restaurante::setNombre(std::string nombre){
    this->nombre=nombre;
}
std::string Restaurante::getNombre(){
    return this->nombre;
}

void Restaurante::setDireccion(std::string direccion){
    this->direccion=direccion;
}
std::string Restaurante::getDireccion(){
    return this->direccion;
}

void Restaurante::setHorario(std::string horario){
    this->horario=horario;
}
std::string Restaurante::getHorario(){
    return this->horario;
}
Restaurante::Restaurante(){} 

File: informe-3/Java/Melchor.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package melchor;

/**
 *
 * @author Ingetelca
 */
public class Melchor {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        Hombre h = new Hombre("Ivan");
        Mujer m = new Mujer();
        Restaurante r = new Restaurante();
        Estudiante e = new Estudiante();
        System.out.println("El nombre del hombre es: " + h.getNombre());
        m.setNombre("Maria");
        m.setEdad(22);
        System.out.println("El nombre de la mujer es: "+ m.getNombre() + " y tiene " 
                            + m.getEdad()+ " años");
        r.setNombre("Pizza pizza");
        r.setDireccion("Zona 9 Guatemala");
        r.setHorario("9:00-23:00");
        System.out.println("El restaurante "+r.getNombre()+" ubicado en "+r.getDireccion()
                           + " atiende en horario de "+r.getHorario());
        e.setNombre("Ivan Melchor");
        e.setCarnet("6190-15-7886");
        e.setNota(76);
        System.out.println("Datos del estudiante: \n "
                + "Nombre: " + e.getNombre() +"\n "
                + "No. Carnet: "+e.getCarnet()+"\n "
                + "Nota Final: "+e.getNota());
    }
       
}
